### DISCUSSION TITLE
Question about how to import csv file into the GUI

### DISCUSSION THREAD

**Discussion by Bokey6** [GiacomoValliPhD/openhdemg#66] [Q&A]
Actually, I tried some times to import .csv file into the GUI. However, everytime I get the same error saying that it did not find any data.
My csv file have 32 columns mains 32 channel. Can you give me some advice to import it? Thanks!
![微信图片_20240517124952](https://github.com/GiacomoValliPhD/openhdemg/assets/110325145/f9f2c074-8409-4bb2-b0ab-d074014f841e)

**Comment by GiacomoValliPhD** [comment]
Hi @Bokey6 

Thanks for using the forum. I can imagine one issue being the absence of decomposition data. Specifically, your .csv file should contain the decomposition outcome (e.g., motor units firing times) as describe here: https://www.giacomovalli.com/openhdemg/tutorials/import_from_other_software/#from-csv-files


In the future, to help you better, you may wish to share some more info about your .csv file (like name of the columns) or also the _openhdemg_ version that you are using. Do not hesitate to reach out again if you need!

**Reply by Bokey6** [reply]
Thanks a lot!
    However, now I just have a raw EMG data, and the size is 28000*32(channel). And the name of the columns is the index of channel. I am a little confused that should I use other software(such as OT BioLab+) to decomposite it first than send it to openhdemg to analyze? I atteched the csv file in the end. Thank you very much for your reply!


[EMG_file.csv](https://github.com/GiacomoValliPhD/openhdemg/files/15349820/EMG_file.csv)

**Reply by GiacomoValliPhD** [reply]
Yes @Bokey6. Currently, _openhdemg_ can be used only for the analysis of motor unit properties from decomposed EMG signals. Therefore, you first need to decomposed your raw signal. 

Please note that _openhdemg_ is changing and expanding constantly. Therefore, additional tools that you may need could become available in the future. If you wish to stay up to date with the future implementations of the project, and see if they will fit better your needs, you can join our newsletter. To do so, just send us an email so that we can include you in the mailing list!

Thanks
